service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/container/CopyAzureStorageContainerBlobsStep.java

```diff
@@ -3,13 +3,90 @@
 import bio.terra.stairway.FlightContext;
 import bio.terra.stairway.Step;
 import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
 import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.AzureStorageAccessService;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.BlobCopier;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.storage.ControlledAzureStorageResource;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.storageContainer.ControlledAzureStorageContainerResource;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import java.util.UUID;
+import org.springframework.http.HttpStatus;
 
-// TODO WOR-591 This step will be responsible for copying files from the
-// source container to the destination
 public class CopyAzureStorageContainerBlobsStep implements Step {
+  private final ResourceDao resourceDao;
+  private final ControlledAzureStorageContainerResource sourceContainer;
+  private final AzureStorageAccessService azureStorageAccessService;
+  private final AuthenticatedUserRequest userRequest;
+  private final BlobCopier blobCopier;
+
+  public CopyAzureStorageContainerBlobsStep(
+      AzureStorageAccessService azureStorageAccessService,
+      ControlledAzureStorageContainerResource sourceContainer,
```

Comment:
For my own learning, how do you determine how you're going to pass parameters to a step (ie via `flightContext` vs step constructor)? Just based on whether the parameter is already present in the flight context when you create the step?

Reply:
Good question. Constructor parameters are only available at flight creation time, i.e., things like spring beans etc. Items created at runtime are passed via the working map.